code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -271,3 +271,82 @@
 	if(L.a_intent == "harm" && L.reagents && !stat)
 		L.reagents.add_reagent("nutriment", 0.4)
 		L.reagents.add_reagent("vitamin", 0.4)
+
+/mob/living/simple_animal/pet/cat/ascendant
+	name = "Ascendant Cat"
+	desc = "It appears this cat has ascended to a godlike form. It'd be best to stay on it's good side."
+	health = 2000
+	maxHealth = 2000
+	unsuitable_atmos_damage = 0
+	var/didcrap = 0
+
+
+/mob/living/simple_animal/pet/cat/ascendant/Life()
+	..()
+	if(stat)
+		return
+	if(health < maxHealth)
+		adjustBruteLoss(-2) //Some life regen.
+	if (didcrap != 1)

```

Comment:
Here is the primary concern, why are you doing this in Life?

You can do this in the .../cat/ascendant/New() after the parent call, i.e

     /mob/living/simple_animal/pet/cat/ascendant/New()
          . = ..() //call parent so we exist and are setup
          if(mind)
                 ... add spells here

That saves this code existing in life and requiring an if check when it only needs to run once, and it also averts the need for a var that you only ever use once.